engine/src/core/events/callback/event_handler.h

```diff
@@ -33,9 +34,14 @@ namespace Bald {
         explicit EventHandler(F&& fun, Args&& ... args) :
                 EventHandlerInterface() {
             if (this->m_ID != 0) {
-                m_Function = [=](const E& ev) {
-                    fun(ev, args...);
-                };
+                if constexpr (std::is_member_function_pointer<F>::value){
```

Comment:
I think EventHandler is not constexpr constructor (it has virtual base class). Let's talk about it.